crates/common/log_manager/src/log_utils.rs

```diff
@@ -93,68 +93,65 @@ pub fn read_log_content(
     }
 }
 
-pub fn filter_logs_on_type(
+pub fn filter_logs(
```

Comment:
This function does what is expected by [#2966](https://github.com/thin-edge/thin-edge.io/issues/2966), but is a bit convoluted.

I would try to use a pipeline style:
- generate a list of `(path, modification_date, wildcard_match)`
- filter out paths out of date range (unless not being the result of a wildcard match)
- sort by modification date
- map to keep only the paths